rero_ils/modules/selfcheck/api.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2020 RERO
+# Copyright (C) 2020 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Selfcheck API."""
+
+from .utils import authorize_selfckeck_user, check_sip2_module, \
+    format_patron_address
+from ..items.api import Item
+from ..libraries.api import Library
+from ..loans.api import Loan, get_loans_by_patron_pid, is_overdue_loan
+from ..patron_transactions.api import PatronTransaction
+from ..patrons.api import Patron
+
+
+def selfcheck_login(login, password):
+    """Selfcheck login handler."""
+    if authorize_selfckeck_user(login, password):
+        staffer = Patron.get_patron_by_email(login)
+        return {
+            'authenticated': staffer.is_librarian,
+            'institution_id': staffer.get_organisation().get('code'),
+            'library_name': Library.get_record_by_pid(
+                staffer.library_pid).get('name')
+        }
+
+
+def validate_patron_account(barcode=None):
+    """Validate patron account handler."""
+    patron = Patron.get_patron_by_barcode(barcode)
+    if patron and patron.is_patron:
+        return True
```

Comment:
At this time your function return either `True` either `None` ? Event if `None` is translate by python as False, isn't it better to return only boolean ?
```
return patron and patron.is_patron
```

Reply:
ok